### PR TITLE
Refactor the traversing of the IMC state machine

### PR DESCRIPTION
**Pull request by Ishadijcks**
Should be tested on the exoskeleton/testjoint.

Provides a much cleaner way to switch between states :)

### CODE REVIEW

File: march_hardware/include/march_hardware/IMotionCubeTargetState.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+namespace march4cpp
+{
+
+class IMotionCubeTargetState
+{
+public:
+  static const IMotionCubeTargetState SWITCH_ON_DISABLED;
+  static const IMotionCubeTargetState READY_TO_SWITCH_ON;
+  static const IMotionCubeTargetState SWITCHED_ON;
+  static const IMotionCubeTargetState OPERATION_ENABLED;
```
Comment (martijnvandermarel) [inline march_hardware/include/march_hardware/IMotionCubeTargetState.h:15]: Doesn't it cause recursive behavior to have objects as attributes in their own class definition? If it works at all? Seems like it should be outside the class, or I am missing some knowledge on cpp that allows this

**Review by martijnvandermarel — COMMENTED** [review]
Nice refactor, see comment

File: march_hardware/include/march_hardware/IMotionCubeTargetState.h (hunk 1 above)
Comment (Ishadijcks) [reply march_hardware/include/march_hardware/IMotionCubeTargetState.h:15]: I see where you are coming from, but here I only 'declare' them. Basically telling the compiler that they exist. Also note they are static (shared among all instances of the class)

We later instantiate them from the .cpp file, which actually creates the objects with a 1 level recursion indeed. However the recursed objects are not yet instantiated, only declared. So `IMotionCubeTargetState::SWITCH_ON_DISABLED::SWITCH_ON_DISABLED` exists, but is only declared so a null pointer.


Implementation based on this question:
https://stackoverflow.com/questions/1965249/how-to-write-a-java-enum-like-class-with-multiple-data-fields-in-c

Explanation based on my current knowledge of cpp, not guaranteed to be correct :')

**Review by JitskedeVries — APPROVED** [review]
Works on test joint

**Pull request merged**